wordpress-mobile/WordPress-iOS
# Local Weekly Roundup

Implements the Weekly Roundup for iOS (with Local Notifications).

WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/438

## Things to do in upcoming PRs:

- Add configuration option to turn these notifications ON / OFF.

## To test:

### Test how the notifications look like and how they react to taps:

1. Check the debug menu at Me > App Setting > Debug > Weekly Roundup
2. Tap on schedule immediately (the other options can be used too but they don't guarantee a time of delivery).  Keep in mind the static fallback / static notification is currently disabled so it will never show at this time.
3. Background the App, and see that the notifications come up.
4. Tap on any of them.

### Test that the trigger time is right:

1. Place a breakpoint at `BackgroundTasksCoordinator.swift:149`.
2. Run the App.
3. When the BP is hit, `po nextDate` and make sure it's next Monday at 10 am (device time).
4. Try this also using another calendar locale.

### Test tracking:

In your console, filter the messaged by "Tracked:".

1. Check the debug menu at Me > App Setting > Debug > Weekly Roundup
2. Tap "Schedule Immediately".
3. You should see up to a maximum of 5 messages that look like this:

🔵 Tracked: notification_scheduled <notification_type: weekly_roundup, site_id: 105284648>

4. Tap on one of the notifications, the App will open, and you should see a tracked notification like this:

🔵 Tracked: notification_tapped <notification_type: weekly_roundup, site_id: 105284648 >

Make sure you also see these events in the live analytics tool.

## Regression Notes

1. Potential unintended areas of impact

None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
